flake.go

```diff
@@ -125,6 +125,10 @@ func getHostId() (uint64, error) {
 	}
 
 	a := addrs[0].To4()
+	if a == nil {
+		a = addrs[0].To16()
```

Comment:
the issue with this code is that you take first 4 bytes of IP address which is a prefix and may be the same for whole subnet. It makes sense to take last 4 bytes instead (interface ID includes them)